Fix reading har with empty response content

dmm opened the pull request:
#### Description

I exported a mitmproxy flow file to a har file but discovered that mitmproxy couldn't read the resulting har. The problem appears to be a response with an empty content. The empty byte string gets passed to http.encoding.encode but it can't encode it, resulting in an unhandled exception.

I added a har test file that reproduces the problem and a simple fix.

#### Checklist

 - [x] I have updated tests where applicable.
 - [ ] I have added an entry to the CHANGELOG.

mhils reviewed (changes requested):
Thanks for the PR! The problem here isn't `b""`, the problem here is an invalid `Content-Encoding` header. Compare https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding with what you have in the example.

Long story short, we should fall back gracefully for all invalid content-encodings, not just for empty bodies.

dmm commented:
Thanks for the feedback! I've renamed the test files to and updated the fix to instead check the provided "content-encoding" value to those supported by "mitmproxy.net.encoding", otherwise falling back to "identity". 

I'm investigating the failing tests (test_har_to_flow[safari] and test_har_to_flow[firefox]) but they seem to be failing for me on main, as well?

mhils commented on mitmproxy/io/har.py:
```diff
@@ -109,7 +120,7 @@ def request_to_flow(request_json: dict) -> http.HTTPFlow:
 
     # Then encode the content, as in `Response.set_content`
     response_content = http.encoding.encode(
-        response_content, response_headers.get("content-encoding") or "identity"
+        response_content, fix_content_encoding(response_headers.get("content_encoding"))
     )
```
I'd like us to be a bit more Pythonic here and ask for forgiveness rather than for permission. Can we just do something like this please?

```python3
    try:
        # Then encode the content, as in `Response.set_content`
        response_content = http.encoding.encode(
            response_content, response_headers.get("content-encoding") or "identity"
        )
    except ValueError:
        logger.warn(f"HAR import: Ignoring invalid content-encoding in request to {request_url!r}")
```

mhils reviewed (commented):
> I'm investigating the failing tests (test_har_to_flow[safari] and test_har_to_flow[firefox]) but they seem to be failing for me on main, as well?

CI is passing, so this needs more details. My only suspicion would be different line endings on Windows vs Linux, but that's just a guess.
